packages/components/psammead-include/src/index.jsx

```diff
@@ -0,0 +1,24 @@
+import React, { useState, useLayoutEffect } from 'react';
+import ReactDOM from 'react-dom';
+
+const Include = ({ html }) => {
+  const [initialised, setInitialised] = useState(false);
+  const [ref, setRef] = useState(React.createRef());
+  const [shadowRoot, setShadowRoot] = useState();
+
+  useLayoutEffect(() => {
+    setShadowRoot(ref.current.parentNode.attachShadow({ mode: 'open' }));
+    setInitialised(true);
+  }, [ref, setInitialised]);
+
+  if (!initialised) {
+    return <div ref={ref}></div>;
+  } else {
+    return ReactDOM.createPortal(
+      <div dangerouslySetInnerHTML={{ __html: html }}></div>,
```

Comment:
This element is inserted into a shadow DOM, but you cannot just pass in raw html, it must be a react element. This *may* render this approach a non-starter 😢